phoenix-core/src/main/java/org/apache/phoenix/hbase/index/IndexRegionObserver.java

```diff
@@ -719,6 +722,23 @@ public void postBatchMutateIndispensably(ObserverContext<RegionCoprocessorEnviro
           for (RowLock rowLock : context.rowLocks) {
               rowLock.release();
           }
+          // Sleep for one millisecond if we have done the data table updates in less than 1 ms. The sleep is necessary
+          // not to allow another data table write on the same row within the same ms. The sleep is very rare to happen.
+          // Assume that a data table write completes at timestamp t.  Now assume another write happens on the same data
+          // row at timestamp t+1. The mutation for unverifying the index table row(s) for the previous write will have
```

Comment:
Nit: Repharse, this as " On second update to the row, 
the mutation for unverifying the index table row(s) for the previous write ....